pkg/controller/rest/vdri/operation_test.go

```diff
@@ -87,6 +135,148 @@ func TestOperation_CreatePublicDID(t *testing.T) {
+		require.NoError(t, err)
+		require.NotEmpty(t, buf)
+
+		require.Equal(t, http.StatusBadRequest, code)
+		verifyError(t, vdri.SaveDIDErrorCode, "parse did doc", buf.Bytes())
+	})
+}
+
+func TestGetDID(t *testing.T) {
+	t.Run("test get did - success", func(t *testing.T) {
+		s := make(map[string][]byte)
+		s["did:peer:21tDAKCERh95uGgKbJNHYp"] = []byte(doc)
+
+		cmd, err := New(&mockprovider.Provider{
+			StorageProviderValue: &mockstore.MockStoreProvider{Store: &mockstore.MockStore{Store: s}},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, cmd)
+		fmt.Println(base64.StdEncoding.EncodeToString([]byte("http://example.edu/credentials/1989")))
+
+		handler := lookupHandler(t, cmd, getDIDPath, http.MethodGet)
+		buf, err := getSuccessResponseFromHandler(handler, nil, fmt.Sprintf(`%s/%s`,
+			vdriDIDPath, base64.StdEncoding.EncodeToString([]byte("did:peer:21tDAKCERh95uGgKbJNHYp"))))
+		require.NoError(t, err)
+
+		response := documentRes{}
+		err = json.Unmarshal(buf.Bytes(), &response)
+		require.NoError(t, err)
+
+		// verify response
+		require.NotEmpty(t, response)
+	})
+
+	t.Run("test get vc - error", func(t *testing.T) {
+		s := make(map[string][]byte)
+		s["did:peer:21tDAKCERh95uGgKbJNHYp"] = []byte(doc)
+
+		cmd, err := New(&mockprovider.Provider{
+			StorageProviderValue: &mockstore.MockStoreProvider{Store: &mockstore.MockStore{Store: s}},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, cmd)
+
+		handler := lookupHandler(t, cmd, getDIDPath, http.MethodGet)
+		buf, code, err := sendRequestToHandler(handler, nil, fmt.Sprintf(`%s/%s`, vdriDIDPath, "abc"))
+		require.NoError(t, err)
+		require.NotEmpty(t, buf)
+
+		require.Equal(t, http.StatusBadRequest, code)
+		verifyError(t, vdri.InvalidRequestErrorCode, "illegal base64 data", buf.Bytes())
+	})
+}
+
+func TestGetDIDRecords(t *testing.T) {
+	t.Run("test get did records", func(t *testing.T) {
+		cmd, err := New(&mockprovider.Provider{
+			StorageProviderValue: mockstore.NewMockStoreProvider(),
+		})
+		require.NoError(t, err)
+		require.NotNil(t, cmd)
+
+		didReq := vdri.DIDArgs{
+			Document: vdri.Document{DID: json.RawMessage(doc)},
+			Name:     sampleDIDName,
+		}
+		jsonStr, err := json.Marshal(didReq)
+		require.NoError(t, err)
+
+		handler := lookupHandler(t, cmd, saveDIDPath, http.MethodPost)
+		buf, err := getSuccessResponseFromHandler(handler, bytes.NewBuffer(jsonStr), handler.Path())
+		require.NoError(t, err)
+		require.NotEmpty(t, buf)
+
+		handler = lookupHandler(t, cmd, getDIDRecordsPath, http.MethodGet)
+		buf, err = getSuccessResponseFromHandler(handler, nil, getDIDRecordsPath)
+		require.NoError(t, err)
+
+		var response didRecordResult
+		err = json.Unmarshal(buf.Bytes(), &response)
+		require.NoError(t, err)
+
+		// verify response
+		require.NotEmpty(t, response)
+		require.Equal(t, 1, len(response.Result))
+	})
+}
+
+func lookupHandler(t *testing.T, op *Operation, path, method string) rest.Handler {
```

Comment:
may be we can make existing lookup function(line 280) to generic rather than writing new one.

Reply:
done